File: leetcode/rust/count-the-number-of-consistent-strings.rs
impl Solution1 {
    pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
        use std::collections::HashSet;

        let mut result = 0;
        let set: HashSet<char> = allowed.chars().collect();

        for word in words {
            let mut is_allowed = true;

            for c in word.chars() {
                if !set.contains(&c) {
                    is_allowed = false;
                    break;
                }
            }

            if is_allowed {
                result += 1;
            }
        }

        result
    }
}

impl Solution2 {
    pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
        use std::collections::HashSet;

        let mut result = 0;
        let set: HashSet<char> = allowed.chars().collect();

        for word in words {
            if word.chars().all(|c| set.contains(&c)) {
                result += 1;
            }
        }

        result
    }
}

File: leetcode/python/count-the-number-of-consistent-strings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: list[str]) -> int:
        allowed_set = {c for c in allowed}
        result = 0

        for word in words:
            if all(c in allowed_set for c in word):
                result += 1

        return result

class Solution1:
    def countConsistentStrings(self, allowed: str, words: list[str]) -> int:
        allowed_set = set(allowed)
        result = 0

        for word in words:
            if set(word) <= allowed_set:
                result += 1

        return result

class Solution2:
    def countConsistentStrings(self, allowed: str, words: list[str]) -> int:
        allowed_set = set(allowed)
        result = 0

        for word in words:
            if allowed_set.issuperset(word):
                result += 1

        return result